src/popup/components/ConnectWalletForm.tsx

```diff
@@ -153,23 +159,19 @@ export const ConnectWalletForm = ({
       }
       saveValue('amount', error ? value : amountValue);
     },
-    [saveValue, currencySymbol, t],
+    [saveValue, currencySymbol, toErrorInfo],
   );
 
-  const handleSubmit = async (ev: React.FormEvent<HTMLFormElement>) => {
-    ev.preventDefault();
-    if (!walletAddressInfo) {
-      setErrors((_) => ({ ..._, walletAddressUrl: 'Not fetched yet?!' }));
-      return;
-    }
+  const handleSubmit = async (ev?: React.FormEvent<HTMLFormElement>) => {
+    ev?.preventDefault();
 
     const errWalletAddressUrl = validateWalletAddressUrl(walletAddressUrl);
     const errAmount = validateAmount(amount, currencySymbol.symbol);
     if (errAmount || errWalletAddressUrl) {
       setErrors((_) => ({
         ..._,
-        walletAddressUrl: errWalletAddressUrl ? t(errWalletAddressUrl) : '',
-        amount: errAmount && t(errAmount),
+        walletAddressUrl: toErrorInfo(errWalletAddressUrl),
+        amount: toErrorInfo(errAmount),
```

Comment:
We are mixing errors as exceptions and errors as value with the `validateWalletAddressUrl` and `validateAmount`. Not a big fan, but I understand why this approach.

It might be worth looking into [neverthrow](https://github.com/supermacro/neverthrow) later on.